tests/python/multidevice/test_dtensor.py

```diff
@@ -204,22 +206,25 @@ def assert_close(expected_tensor, dtensor):
 
 @pytest.mark.mpi
 def test_row_parallel_linear(setup_default_process_group, multidevice_test):
-    d, b, s, e = dist.get_world_size(), 2, 1024, 768
+    d, b, s, e = dist.get_world_size(), 2, 3, 5
     rank = dist.get_rank()
     torch.cuda.set_device(rank)
 
     mesh = dist.device_mesh.init_device_mesh("cuda", [d])
 
-    inp_tensor = torch.randn(b, s, d * e, requires_grad=True)
-    weight_tensor = torch.randn(e, d * e, requires_grad=True)
+    inp_tensor = torch.randint(
```

Comment:
I understand the motivation here.
Did you attempt scaling the parameters as done in these tests: https://github.com/NVIDIA/Fuser/blob/697c696260673efe81c0c6607b86adefd81f1241/tests/cpp/test_multidevice_transformer.cpp#L29C1-L32C44

These tests use bfloat16

Reply:
The sizes in this test are not GPT2 sizes, so I'm not sure what scale to use. Besides, randn in the old code generates a normal distribution, whose range is already pretty small. Further scaling down might help but the test would be fragile depending on a magic value.